dali/pipeline/graph/op_graph_verifier.cc

```diff
@@ -116,20 +116,10 @@ void CheckArgumentInputConstraints(const OpGraph& op_graph, const OpNode& op) {
                  to_string(op.op_type) + " Ops do not support tensor arguments, found in " +
                      op.instance_name + " Op.");
   }
-  for (const auto& arg_pair : op.spec.ArgumentInputs()) {
```

Comment:
What happens if you pass a GPU node here?

Reply:
It will fail on this check:
```
DALI_ENFORCE(it->second.has_cpu, "cpu input requested by op exists "
                            "only on GPU. " + error_str);
```
from `pipeline.cc`